tests/test_new_return_types_qnode.py

```diff
@@ -0,0 +1,1665 @@
+            qml.Hadamard(wires=[0])
+            qml.CRX(x, wires=[0, 1])
+            return qml.apply(measurement1), qml.apply(measurement2)
+
+        qnode = qml.QNode(circuit, dev, diff_method=None)
+        res = qnode(0.5)
+
+        qml.disable_return()
+
+        all_shot_copies = [
+            shot_tuple.shots for shot_tuple in dev.shot_vector for _ in range(shot_tuple.copies)
+        ]
+
+        assert len(res) == len(all_shot_copies)
+        for r, shots in zip(res, all_shot_copies):
+
+            shape = () if shots == 1 else (shots,)
+            assert all(res_item.shape == shape for res_item in r)
+
+    @pytest.mark.parametrize("measurement1", [qml.counts(qml.PauliZ(0)), qml.counts(wires=[0])])
+    @pytest.mark.parametrize("measurement2", [qml.counts(qml.PauliZ(0)), qml.counts(wires=[0])])
+    def test_counts(self, shot_vector, measurement1, measurement2, device):
+        """Test multiple counts measurements."""
+        qml.enable_return()
+
+        dev = qml.device(device, wires=2, shots=shot_vector)
+
+        def circuit(x):
+            qml.Hadamard(wires=[0])
+            qml.CRX(x, wires=[0, 1])
+            return qml.apply(measurement1), qml.apply(measurement2)
+
+        qnode = qml.QNode(circuit, dev, diff_method=None)
+        res = qnode(0.5)
+
+        qml.disable_return()
+
+        all_shots = sum([shot_tuple.copies for shot_tuple in dev.shot_vector])
+
+        assert isinstance(res, tuple)
+        assert len(res) == all_shots
+        for r in res:
+            assert isinstance(r, tuple)
+            assert all(isinstance(res_item, dict) for res_item in r)
+
+
+# -------------------------------------------------
+# Shot vector multi measurement tests - test data
+# -------------------------------------------------
+
+pauliz_w2 = qml.PauliZ(wires=2)
+proj_w2 = qml.Projector([1], wires=2)
+hermitian = qml.Hermitian(np.diag([1, 2]), wires=0)
+tensor_product = qml.PauliY(wires=2) @ qml.PauliX(wires=1)
+
+# Expval/Var with Probs
+
+scalar_probs_multi = [
+    # Expval
+    (qml.expval(pauliz_w2), qml.probs(wires=[2, 0])),
+    (qml.expval(proj_w2), qml.probs(wires=[2, 0])),
+    (qml.expval(tensor_product), qml.probs(wires=[2, 0])),
+    # Var
+    (qml.var(qml.PauliZ(wires=1)), qml.probs(wires=[0, 1])),
+    (qml.var(proj_w2), qml.probs(wires=[2, 0])),
+    (qml.var(tensor_product), qml.probs(wires=[2, 0])),
+]
+
+# Expval/Var with Sample
+
+scalar_sample_multi = [
+    # Expval
+    (qml.expval(pauliz_w2), qml.sample(op=qml.PauliZ(1) @ qml.PauliZ(0))),
+    (qml.expval(proj_w2), qml.sample(op=qml.PauliZ(1) @ qml.PauliZ(0))),
+    (qml.expval(tensor_product), qml.sample(op=qml.PauliZ(0))),
+    # Var
+    (qml.var(proj_w2), qml.sample(op=qml.PauliZ(1) @ qml.PauliZ(0))),
+    (qml.var(pauliz_w2), qml.sample(op=qml.PauliZ(1) @ qml.PauliZ(0))),
+    (qml.var(tensor_product), qml.sample(op=qml.PauliZ(0))),
+]
+
+scalar_sample_no_obs_multi = [
+    # TODO: for copy=1, the wires syntax has a bug
```

Comment:
> the wires syntax has a bug

Should this be no wires syntax? :thinking: I think I might have written this comment, but can't recollect. :laughing:

Reply:
Yes I don't know what it referred to haha let me check that